# Experimental evolution of Bacteroides uniformis under xenobiotic exposure

The human gut microbiota routinely encounters thousands of foreign compounds, referred to as xenobiotics. While many of these compounds are known to affect gut bacterial growth, whether and how they shape the genetic adaptation of gut bacteria remains largely unknown. Due to the complexities and multiple confounders impacting gut microbial evolution in vivo, we used a controlled laboratory evolution approach to study the genetic variation of Bacteroides uniformis, a prevalent and abundant member of the human gut microbiota, under exposure to 37 xenobiotics that are known to be encountered by human gut microbiota. These include 24 medications, six environmental contaminants, four food additives, and three natural food compounds. Only three medications show significant growth inhibition in the beginning of the experiments and those inhibition effects were eliminated by the end of the adaptive evolution. Whole genome sequencing of the lineages after circa 100 generations identified various genomic regions with higher evolvability. Notably, variations in the coding region as well in the upstream regulatory region of SusC/D genes were found under multiple xenobiotic exposures. Moreover, xenobiotic specific genetic mutations were found within the coding sequences of the Resistance-Nodulation-Division (RND) efflux pump for ultra-high concentration PFAS, related to transcriptional regulator AcrR when evolving with xanthan gum, and the potassium uptake system protein TrkAH coding genes with loperamide exposure. These mutations may indicate changes in their ability to utilize fibres and transport chemical compounds across membranes. To assess the phenotypic influence of the observed genetic changes, we examined the growth of evolved isolates to multiple antibiotic resistance and observed a four-times increased resistance towards erythromycin in xanthan gum evolved populations. These results deepen our understanding of the gut microbiome's adaptation to xenobiotic exposure with implications for understanding microbiome variation and for preventing emergence of antimicrobial resistance.

# Data and code

This repository contains raw growth data, vcf files generated after running _snippy_ pipeline, as well as scripts for data visulization.

Raw sequencing data was uploaded to European Nucleotide Archive with public access under session ID PRJEB72819 (including Nanopore sequencing reads of parental strain and Illumina NextSeq genomic sequencing reads of evolved populations) and PRJEB72794 (Illumina NovaSeq 6000 genomic sequencing short reads of populations exposed under extra-high concentration PFAS, reported in Lindell et al., 2024).
