---
title: "Gene wise variant analysis"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_depth: 4
date: "2023-10-26"
---

```{r setup, include=FALSE}
  knitr::opts_chunk$set(echo = TRUE)
  source("settings.R")
  map <- read.table("../00.data/vcf_20231024_effect_map.txt", 
                    header = T, sep = "\t")
  c_shape <- map$Shape
  names(c_shape) <- map$Effect
  
  t_shape <- map$Effect_type
  names(t_shape) <- map$Effect
```

### Overview of variants 

We firstly checked the most prevalent (appeared in more than 50 populations) and abundant variants (average allele frequency > 0.2) variants.

```{r p_overview, echo = FALSE, warning = FALSE, fig.height=12, fig.width=16}
# read in all variants detected in ALE 1.0 and ALE 2.0 with AF > 0.05
  all <- readRDS("../03.results/20231023_all_005.rds")
  pos_gene <- read.table("../00.data/vcf_20231025_position_gene_map.txt",
                         header = F, sep = "\t")
  colnames(pos_gene) <- c("POS", "Type", "Gene")
  all <- merge(all, pos_gene)
  all$Effect <- as.factor(all$Effect)

  t1 <- all %>% group_by(Gene) %>% summarise(Prevalence =
                                            length(unique(Sample_ID)),
                                            Average_AF = mean(Ratio))
  
  top_gene <- t1[(t1$Prevalence > 50) | (t1$Average_AF > 0.2), ]
  
  top <- all[all$Gene %in% top_gene$Gene, ]
  top_lst <- unique(top$Gene)
  
  ggplot(top, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect)) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Gene, levels = top_lst), 
               ncol = 4, scales = "free_y") +
    theme(legend.position = "top",
          axis.text.x = element_text(size = 8, angle = 90, vjust = 0.5, hjust=1))
```

### Missense variants in gene peg.962

For gene peg.962 (__RND efflux system inner membrane transporter__), multiple variants (N = 9) were detected in six different positions in ALE2.0, with most of them (seven out of nine) dominate the populations (with allele frequency higher than 0.9).

```{r p962, echo = F, warning=F, fig.width=5, fig.height=3}
  this_gene_ID <- "peg.962"

  this_gene <- all[all$Gene == this_gene_ID, ]
  gff <-read.delim("../00.data/annot_RAST_simp.gff", header = F, sep = "\t")
  colnames(gff) <- c("Start", "End", "Strand", "Gene_ID", "Annotation")
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID == this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID == this_gene_ID, ]$End
  
  ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Compound, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = c("PFNA" = "lightblue4",
                                 "PFOA" = "yellow4")) +
    labs(x = "Position") +
    xlim(this_start, this_end) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") +
    main_theme 
```

### Non-coding region variant before gene peg.1113

The first gene shown as peg.1114, turns out to be a non-coding region variants at position 1,425,628 located in the upstream (d = 40 bp) of gene peg.1113, which is identified as [Lipoprotein](https://www.uniprot.org/uniprotkb/A7V0G6) in [UniProt](https://www.uniprot.org/), which is confirmed by the [String](https://version-11-5.string-db.org/network/411479.BACUNI_01051) as it neighbored coding gene of Fructokinase (peg. 1114). In _B. uniformis_ genome, in total, two copies of Fructokinase gene were annotated (the other is peg. 2097).

```{r peg1113, echo=FALSE, warning=FALSE}
  p1114 <- all[all$Gene == "peg.1114", ]
  #class(p1114$Effect)
  ggplot(p1114, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
```

### Variants around gene peg.1246/1250

For mutations around gene peg.1246 and peg.1250, as we could see, most of them are within the coding region, with only 2 in non-coding region. For gene peg.1246 (Fic family protein), most of the variants have AF < 0.1, with one frame shift variants (at the position 1,600,695) above slightly. For the second gene (peg.1250), no function is annotated and blasting again UniProt shows alignment to [DUF4468 domain-containing protein](https://www.uniprot.org/uniprotkb/A0A174MTB8/entry). A frame shift variant with AF = 1 is detected in multiple samples, however, no replicate were found for any compound. 

```{r peg1246, echo=FALSE, warning=FALSE, fig.width=8, fig.height=6}
  this_gene_ID <- c("peg.1246", "peg.1250")

  this_gene <- all[all$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_gff <- gff[gff$Gene_ID %in% this_gene$Gene, ]
  this_start <- min(this_gff$Start)
  this_end <- max(this_gff$End)
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    labs(x = "Position", y = "Allele frequency") +
    scale_size(range = c(2.5, 5)) +
    geom_vline(xintercept = c(this_gff$Start, this_gff$End), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    #xlim(this_start, this_end) +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") +
    guides(colour = guide_legend(order = 1, nrow = 2), 
         shape = guide_legend(order = 2, nrow = 2),
         size = guide_legend(order = 3, nrow = 2)) +
    main_theme 
  
  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               position = position_dodge(width=0.3), alpha = 0.9) + 
    main_theme +
    labs(y = "Allele frequency") +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  
  plot_grid(a, b, nrow = 2, align = "v", axis = "l", 
                    rel_heights = c(1.15, 1.2), 
                    labels = c('a', 'b'))
```

### Variants around gene peg.1351

Variants before and within the Glutamyl-tRNA synthetase gene (peg.1351) show a low allele frequency in general. Only two missense variants were detected within the coding region (in Montelukast and Acarbose) at very close positions in the end of the protein sequence (501/502 out of 508 amino acids).

```{r peg1351, echo=FALSE, warning=FALSE, fig.width=8, fig.height=6}
  this_gene_ID <- "peg.1351"
  this_gene <- all[all$Gene == this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID == this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID == this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, nrow = 2, align = "v", axis = "l", 
                    rel_heights = c(1, 1.2), 
                    labels = c('a', 'b'))
```

### Variants around gene peg.1416/1417

For mutations around gene peg.1416 and peg.1417, both genes are on the + strand, which means, only one synonymous mutation was detected in the end of gene peg.1416 (Histidine ammonia-lyase (EC 4.3.1.3);Ontology_term=KEGG_ENZYME:4.3.1.3). For gene peg.1417, which is annotated as __transcriptional regulator AcrR family__, which is reported to be a regulation system that confers resistance to the antibiotic tetracycline [Kang et al., 2019](https://febs.onlinelibrary.wiley.com/doi/full/10.1002/2211-5463.12710) and regulate the AcrAB-TolC efflux pump (peg. 1418, 1419, and 1420). Notably, variants before and within gene AcrR are mainly detected in Xanthan gum evolved samples. All populations (N = 8 replicates with two different concentrations of Xanthan gum) have either non coding region or frame shift variant detected.

```{r peg1416, echo=FALSE, warning=FALSE, fig.width=10, fig.height=5}
  this_gene_ID <- c("peg.1416", "peg.1417")

  this_gene <- all[all$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_gff <- gff[gff$Gene_ID %in% this_gene$Gene, ]
  this_start <- min(this_gff$Start)
  this_end <- max(this_gff$End)
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    labs(x = "Position", y = "Allele frequency") +
    scale_size(range = c(2.5, 5)) +
    geom_vline(xintercept = c(this_gff$Start, this_gff$End), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    #xlim(this_start, this_end) +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") +
    main_theme 
  
  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               position = position_dodge(width=0.3), alpha = 0.9) + 
    main_theme +
    labs(y = "Allele frequency") +
    scale_color_manual(values = concentration_color) +
    scale_size(range = c(2.5, 5)) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", rel_widths = c(1.3, 1),
                    labels = c('a', 'b'))
```

### Variants between 1912 - 1945 kb ()

```{r region_1912_1945, echo=FALSE, warning=FALSE, fig.width=10, fig.height=8}
  ### get region from peg.1535 to peg.1563
  region <- all[(all$POS > 1910551) & (all$POS < 1944637), ]

  ## upper panel: ratio of variants and prevalence of variants
  t1 <- region %>% group_by(POS) %>% summarise(Prevalence =
                                              length(unique(Sample_ID)),
                                            Average_AF = mean(Ratio))
  t1_des <- unique(all[, c("POS", "FTYPE")])
  t1 <- merge(t1, t1_des)
  
  p1_1 <- ggplot(t1, aes(POS, Prevalence)) +
    geom_point(aes(color = FTYPE, size = Average_AF),
               alpha = 0.7, shape = 16) + 
    scale_color_manual(values = c("gold", "gray47")) +
    xlab("") +
    scale_size(range = c(0.25, 5)) +
    theme_minimal_hgrid(color = "gray88", font_size = 10) +
    theme(legend.position = "none") +
    main_theme 
  
  p1_2 <- ggplot(region, aes(POS)) + 
    geom_density(aes(color = FTYPE, y = ..scaled..)) +
    scale_color_manual(values = c("gold", "gray47")) +
    scale_y_continuous(position = "right") +
    xlab("") + ylab("Density of variants") +
    theme_minimal_hgrid(color = "gray88", font_size = 10) +
    theme(legend.position = "top",
        legend.justification = c("left", "center")) +
    main_theme
  
  aligned_plots <- align_plots(p1_2, p1_1, align="hv", axis="tblr")
  p1 <- ggdraw(aligned_plots[[2]]) + draw_plot(aligned_plots[[1]])
  ggsave("upper.pdf", p1, width = 10, height = 2.5)
  
  ### second panel for the gene plot
  library(gggenes)
  gene_lst <- paste0("peg.", seq(1535, 1563))
  region_gff <- gff[gff$Gene_ID %in% gene_lst, ]
  colnames(region_gff) <- c("start", "end", "strand", "gene", "protein")
  
  region_gff$orientation <- ifelse(region_gff$strand == "+", 1, 0)
  region_gff$molecule <- "Genes"
  
  p2 <- ggplot(region_gff, aes(xmin = start, xmax = end,
                  y = molecule, fill = protein, 
                  forward = orientation)) +
    geom_gene_arrow() + labs(y = "") +
    #facet_wrap(~ molecule, scales = "free", ncol = 1) +
    scale_fill_brewer(palette = "Set3") +
    theme_genes() +
      theme(legend.position = "none")
  ggsave("middle.pdf", p2, width = 10, height = 3)
  ## bottom panel, show the AF of each variants from each compounds
  p3 <- ggplot(region, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect,
                   size = Ratio), alpha = 0.8) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    scale_color_manual(values = c("0" = "gray",
                                "25" = "#a0cbe8", 
                                "250" = "navyblue",
                                "50" = "#1170aa", 
                                "500" = "navyblue")) +
    
    scale_size(range = c(0.25, 5)) +
    labs(x = "", y = "Allele frequency") +
    main_theme +
    theme(legend.position = "bottom", 
          legend.justification = c("left", "center")) +
    guides(color = guide_legend(nrow = 2),
           shape = guide_legend(nrow = 2),
           size = guide_legend(nrow = 2))
  
  ggsave("lower.pdf", p3, width = 10, height = 3.5)
  p <- plot_grid(p1, p3, nrow = 2, rel_heights = c(1, 1),
                 align = "v", axis = "lr")
  
  #align = "v", axis = "lr",
  p
```

### Variants around gene peg.1537

Variants before the gene (peg.1537) were detected at four different positions (1914983, 1914985, 1914986, and 1914999), among those, variants at three positions were also detected in parental populations, therefore, we assume these variants might not affect the growth or resistance of the population.

```{r peg1537, echo=FALSE, warning=FALSE, fig.width=8, fig.height=6}
  this_gene_ID <- "peg.1537"
  this_gene <- all[all$Gene == this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID == this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID == this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, nrow = 2, align = "v", axis = "l", 
                    rel_heights = c(1, 1.2), 
                    labels = c('a', 'b'))
```


### Variants around gene peg.1554/1555

For mutations within gene peg.1554 (Integrase) and peg.1555 (Outer membrane TonB-dependent transporter utilization system for glycans and polysaccharides/PUL SusC family), variants are concentrated in several positions. Frame shift variant was detected at the position 1,931,428 within the integrase gene, with the same variant detected in 1B9 (one of the parental population). 

For TonB gene, only synonymous and missense variants were detected.

```{r peg1554, echo=FALSE, warning=FALSE, fig.width=8, fig.height=6}
  this_gene_ID <- c("peg.1554", "peg.1555")

  this_gene <- all[all$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_gff <- gff[gff$Gene_ID %in% this_gene$Gene, ]
  this_start <- min(this_gff$Start)
  this_end <- max(this_gff$End)
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    labs(x = "Position", y = "Allele frequency") +
    scale_size(range = c(2.5, 5)) +
    geom_vline(xintercept = c(this_gff$Start, this_gff$End), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    #xlim(this_start, this_end) +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") +
    main_theme 
  
  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               position = position_dodge(width=0.3), alpha = 0.9) + 
    main_theme +
    labs(y = "Allele frequency") +
    scale_color_manual(values = concentration_color) +
    scale_size(range = c(2.5, 5)) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  
  plot_grid(a, b, nrow = 2, align = "v", axis = "l", 
                    rel_heights = c(1.15, 1.25), 
                    labels = c('a', 'b'))
```

### Variants before gene peg.2424

Variants were detected before gene peg.2424 (unknown, 68 amino acids), with AF > 0.1 found in higher concentration PFOA (ALE 2.0) and Xanthan gum.

```{r peg2424, echo=FALSE, warning=FALSE, fig.width=9, fig.height=6}
  this_gene_ID <- c("peg.2424")

  this_gene <- all[all$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_gff <- gff[gff$Gene_ID %in% this_gene$Gene, ]
  this_start <- min(this_gff$Start)
  this_end <- max(this_gff$End)
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    labs(x = "Position", y = "Allele frequency") +
    scale_size(range = c(2.5, 5)) +
    #geom_vline(xintercept = c(this_gff$Start, this_gff$End), 
    #           linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    #xlim(this_start, this_end) +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") +
    main_theme 
  
  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               position = position_dodge(width=0.3), alpha = 0.9) + 
    main_theme +
    labs(y = "Allele frequency") +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    scale_size(range = c(2.5, 5)) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  
  plot_grid(a, b, nrow = 2, align = "v", axis = "l", 
                    rel_heights = c(1.15, 1.2), 
                    labels = c('a', 'b'))
```

### Variants before gene peg.2727

Variants were detected before gene peg.2727 (ompA, outer membrane protein, strand -), with AF > 0.1 found in many populations including parental and control samples.

```{r peg2727, echo=FALSE, warning=FALSE, fig.width=9, fig.height=6}
  this_gene_ID <- c("peg.2727")

  this_gene <- all[all$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_gff <- gff[gff$Gene_ID %in% this_gene$Gene, ]
  this_start <- min(this_gff$Start)
  this_end <- max(this_gff$End)
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    labs(x = "Position", y = "Allele frequency") +
    scale_size(range = c(2.5, 5)) +
    geom_vline(xintercept = c(this_gff$Start, this_gff$End), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    #xlim(this_start, this_end) +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") +
    main_theme 
  
  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               position = position_dodge(width=0.3), alpha = 0.9) + 
    main_theme +
    scale_size(range = c(2.5, 5)) +
    labs(y = "Allele frequency") +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  
  plot_grid(a, b, nrow = 2, align = "v", axis = "l", 
                    rel_heights = c(1.15, 1.2), 
                    labels = c('a', 'b'))
```

### Variants around gene peg.2855, 2981, and 3075

Variants were detected around these three genes, with a variant detected in PFNA (ALE 2.0) before gene peg.2981 (UDP-glucose 4-epimerase (EC 5.1.3.2);Ontology_term=KEGG_ENZYME:5.1.3.2) at 3,543,736. 

```{r peg2855, echo=FALSE, warning=FALSE, fig.width=9, fig.height=6}
  this_gene_ID <- c("peg.2855", "peg.2981", "peg.3075")
  this_gene <- all[all$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_gff <- gff[gff$Gene_ID %in% this_gene$Gene, ]
  this_start <- min(this_gff$Start)
  this_end <- max(this_gff$End)
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    labs(x = "Position", y = "Allele frequency") +
    scale_size(range = c(2.5, 5)) +
    #geom_vline(xintercept = c(this_gff$Start, this_gff$End), 
    #           linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    #xlim(this_start, this_end) +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") +
    main_theme 
  
  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               position = position_dodge(width=0.3), alpha = 0.9) + 
    main_theme +
    labs(y = "Allele frequency") +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  
  plot_grid(a, b, nrow = 2, align = "v", axis = "l", 
                    rel_heights = c(1.15, 1.2), 
                    labels = c('a', 'b'))
```

### Variants around gene peg.3135 and peg.3342

For gene peg.3135 (3683196 to 3683321, + strand), pipetide with 42 amino acids, all variants were insertion and detected at the position 3,683,056, 140 bp upstream of peg.3135 and 82 bp downstream of previous gene (peg.3134, Methyl-directed repair DNA adenine methylase (EC 2.1.1.72);Ontology_term=KEGG_ENZYME:2.1.1.72). 

Variants were detected before and within this gene, annotated as Acyl_transf_3 from Acyltransferase family (3929637 to	3929819, + strand).

```{r peg3342, echo=FALSE, warning=FALSE, fig.width=9, fig.height=6}
  this_gene_ID <- c("peg.3342", "peg.3135")
  this_gene <- all[all$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_gff <- gff[gff$Gene_ID %in% this_gene$Gene, ]
  this_start <- min(this_gff$Start)
  this_end <- max(this_gff$End)
  
  ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               position = position_dodge(width=0.3), alpha = 0.9) + 
    main_theme +
    facet_wrap(~Gene, nrow = 2) +
    labs(y = "Allele frequency") +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
```