---
title: "Compound wise variant analysis"
output: 
  html_document:
    toc: true
    toc_float: true
    toc_depth: 4
date: "2023-10-31"
---

```{r setup, include=FALSE}
  knitr::opts_chunk$set(echo = TRUE)
  source("settings.R")
  map <- read.table("../00.data/vcf_20231024_effect_map.txt", 
                    header = T, sep = "\t")
  c_shape <- map$Shape
  names(c_shape) <- map$Effect
  
  t_shape <- map$Effect_type
  names(t_shape) <- map$Effect
  
  gff <-read.delim("../00.data/annot_RAST_simp.gff", header = F, sep = "\t")
  colnames(gff) <- c("Start", "End", "Strand", "Gene_ID", "Annotation")
  
```

### Overview of variants 

After checking the most prevalent & abundant (appeared in more than 50 populations and average allele frequency > 0.2) variants, we also analyzed the variants detected in each compound evolved populations. This is to not miss the variants that are specific to compound (therefore not prevalent) and have varied allel frequency (therefore average <= 0.2 among all detected populations).

```{r cmp_wise, echo = FALSE, warning = FALSE, fig.height=12, fig.width=20}
# read in all variants detected in ALE 1.0 and ALE 2.0 with AF > 0.05
  all <- readRDS("../03.results/20231023_all_005.rds")
  pos_gene <- read.table("../00.data/vcf_20231025_position_gene_map.txt",
                         header = F, sep = "\t")
  colnames(pos_gene) <- c("POS", "Type", "Gene")
  all <- merge(all, pos_gene)
  all$Effect <- as.factor(all$Effect)

  ggplot(all, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect)) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Compound), 
               ncol = 6, scales = "free_y") +
    labs(y = "Allele frequency") +
    theme(legend.position = "top")
```

### Skip list according to gene wise variant results

We filtered out variants around the following gene (as they were appearing in many compounds and discussed in gene wise variant analysis): __peg.962__, detected in ALE 2.0, discussed in gene wise analysis; __peg.1114__ (in fact closer to peg.1113 lipoprotein); __peg.1351__; __peg.1554__ (detected in parental and control samples with AF == 1); __peg.1416, peg.1417__; __peg.1537__; __peg.1554__, and __peg.1555__.

```{r others, echo = FALSE, warning = FALSE, fig.height=16, fig.width=20}
  #skip_lst <- c("peg.962", "peg.1114", "peg.1351", "peg.1554", "peg.1416",
  #              "peg.1417", "peg.1537", "peg.1554", "peg.1555")
  skip_lst <- c("peg.1114", "peg.1351", "peg.1537", "peg.1554", "peg.1555",
                "peg.2727", "peg.3135", "peg.3342")
  all_others <- all[!(all$Gene %in% skip_lst), ]
  
  ggplot(all_others, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect)) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Compound), 
               ncol = 5) +
    labs(y = "Allele frequency") +
    theme(legend.position = "top")
```

### Variants detected in ALE 2.0

We extracted variants in ALE 2.0 and compared them to the PFNA and PFOA in ALE 1.0 (no filter for AF, i.e. all variants with AF > 0.05; or gene).

```{r ALE2, echo = FALSE, warning = FALSE, fig.height=9, fig.width=9}
  ale_cmp <- all[all$Compound %in% c("DMSO", "DMSO_control", "NT5002",
                                              "PFNA", "PFOA"), ]
  ggplot(ale_cmp, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    scale_size(range = c(0.25, 5)) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Compound), 
               nrow = 5, scales = "fixed") +
    labs(y = "Allele frequency") +
    theme(legend.position = "right")
```

#### Variants detected in ALE 2.0 - zoom in peg.319 

Gene peg.319 is annotated as beta-glucosidase (one of the 17 copies in the genome; EC 3.2.1.21;Ontology_term=KEGG_ENZYME:3.2.1.21; Glycosyl hydrolase family 3 C-terminal domain protein OS=Bacteroides uniformis in UniProt).

```{r peg319, echo=FALSE, warning=FALSE, fig.width=8, fig.height=4}
  this_gene_ID <- "peg.319"
  this_gene <- all_others[all_others$Gene == this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID == this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID == this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(1.4, 1), 
                    labels = c('a', 'b'))
```

#### Variants detected in ALE 2.0 - zoom in peg.962 

This has been discussed in gene wise analysis (1200664 - 1203843, strand -).

#### Variants detected in ALE 2.0 - zoom in 1,912 Kb - 1,945 Kb

We then extracted and plot the hot sport region, from 1,912 to 1,945 Kb. Five copies of SusC-SusD found in this region. Gene IDs in this region range from peg.1536 to peg.1563.

As we already analyzed __peg.1537__(actually suppose to be __peg.1536__), __peg.1554__ (detected in parental and control samples with AF == 1), we will filter these two gene out here. 

```{r peg_region, echo=FALSE, warning=FALSE, fig.width=8, fig.height=7}
  this_gene_ID <- paste0("peg.", seq(1536, 1563))

  this_gene_ID <- this_gene_ID[!(this_gene_ID %in% c("peg.1537",
                                                     "peg.1554"))]
  this_gene <- all_others[all_others$Gene %in% this_gene_ID, ]
  
  this_gene <- this_gene[this_gene$Compound %in% c("DMSO", "DMSO_control",
                                                   "NT5002", "PFNA", "PFOA"), ]
  
  ## Get the coordinate of genes with varaints detected within or nearby
  #IDs <- unique(this_gene$Gene)
  IDs <- c("peg.1536", "peg.1544", "peg.1551", "peg.1555", "peg.1562")
  starts <- gff[gff$Gene_ID %in% IDs, ]$Start
  ends <- gff[gff$Gene_ID %in% IDs, ]$End
  this_start <- min(starts)
  this_end <- max(ends)
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_size(range = c(0.125, 2.5)) +
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(starts, ends), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "none") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9, position = position_dodge(width=0.3)) + 
    main_theme +
    scale_size(range = c(0.125, 2.5)) +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    labs(y = "Allele frequency") +
    guides(colour = guide_legend(order = 1, ncol = 3), 
         shape = guide_legend(order = 2, ncol = 2),
         size = guide_legend(order = 3, ncol = 3)) +
    theme(legend.position = "right",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, nrow = 2, align = "v", axis = "l", 
                    rel_heights = c(1, 1), 
                    labels = c('a', 'b'))
```

As we can see, among five copies of TonB, the 2nd (__peg.1544__) and 5th (__peg.1562__) have more variants detected. To zoom in those two genes:

```{r peg_region_2, echo=FALSE, fig.height=4, fig.width=8, warning=FALSE}
  IDs <- c("peg.1544", "peg.1562")
  starts <- gff[gff$Gene_ID %in% IDs, ]$Start
  ends <- gff[gff$Gene_ID %in% IDs, ]$End
  
  df_pos <- data.frame(Gene = c(IDs, IDs),
                       Coord = c(starts, ends))
  
  this_gene_sub <- this_gene[this_gene$Gene %in% IDs, ]
  
 ggplot(this_gene_sub, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_size(range = c(0.125, 2.5)) +
    scale_color_manual(values = concentration_color) +
    facet_grid(~Gene, scales = "free") +
    labs(x = "Position") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "top") + 
    guides(colour = guide_legend(order = 1, nrow = 2), 
         shape = guide_legend(order = 2, nrow = 2),
         size = guide_legend(order = 3, nrow = 2)) +
    labs(y = "Allele frequency") +
    main_theme 
```

#### Variants detected in ALE 2.0 - zoom in peg. 1962

This gene (peg.1962, 2,402,232 to 2,403,446, + strand) is annotated as site-specific recombinase, belonging to thephage integrase family. Variants in non-coding region (at 2,403,658) seem to be located after this gene and before peg.1963 (Capsular polysaccharide transcription antitermination protein __UpxY family__, or __[NusG](https://www.frontiersin.org/articles/10.3389/fmicb.2020.619618/full)__ or __KOW domain-containing protein__; 2,403,943 to 2,404,491). __[NusG](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC8060035/)__ is an intrinsic transcription termination factor that stimulates motility and coordinates gene expression with NusA (peg.1227).

```{r peg1962, echo=FALSE, warning=FALSE, fig.width=10, fig.height=4}
  this_gene_ID <- c("peg.1962", "peg.1963")
  this_gene <- all[all$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID %in% this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID %in% this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(1.4, 1), 
                    labels = c('a', 'b'))
```

#### Variants detected in ALE 2.0 - zoom in peg.2727

This has been discussed in gene wise analysis (3255197 to 3256387, strand -, hypothetical protein).

#### Variants detected in ALE 2.0 - region after 3.5M 

There are few mutations with AF > 0.5 after 3,500 Kb, however, non of those has replicates. Among which, we have plotted peg.3135 and peg.3342, will be skip here.

```{r peg_region_3.5M, echo=FALSE, warning=FALSE, fig.width=6, fig.height=6}
  this_region <- all[all$POS > 3500000, ]
  #this_region <- this_region[this_region$Ratio > 0.1, ]
  this_region <- this_region[!(this_region$Gene %in% c("peg.3135", "peg.3342")), ]
  this_region_05 <- this_region[this_region$Ratio > 0.5, ]
  this_region <- this_region[this_region$Gene %in% this_region_05$Gene, ]
  
 ggplot(this_region, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_size(range = c(0.125, 2.5)) +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~Gene, scales = "free") +
    labs(x = "Position") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1)) + 
    labs(y = "Allele frequency") +
    main_theme 
```

```{r ALE2_01, eval = FALSE, echo = FALSE, warning = FALSE, fig.height=7, fig.width=7}
  ## We filtered out variants with AF > 0.1, as below.
  #### Variants with AF > 0.1 - PFAS only
  ale_01 <- ale_cmp[ale_cmp$Ratio > 0.1, ]

  ggplot(ale_01, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    scale_size(range = c(0.5, 5)) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Compound), 
               scales = "fixed") +
    labs(y = "Allele frequency") +
    theme(legend.position = "right")
```

### Variants detected for compound Loperamide (following Figure 1a)

Firstly, we checked the overall distribution of variants in compound and control populations.

```{r get_lop, echo = FALSE, warning=FALSE}
  lop <- all_others[all_others$Compound %in% c("DMSO_control", "NT5002",
                                              "Loperamide"), ]
  lop_02 <- lop[lop$Ratio > 0.2, ]
```

Among those variants, we decided to focus on the areas where AF > 0.2 variants were detected, including coding area `r unique(lop_02$Gene)`. There are two non-coding region variants with AF ~ 0.18 around gene peg.393 (Two-component system sensor histidine kinase) and peg.538 (Outer membrane TonB-dependent transporter utilization system for glycans and polysaccharides (PUL) SusC family), however, no replicates were found in those region. 

```{r lop, echo=FALSE, warning=FALSE, fig.width=8, fig.height=6}
   ggplot(lop, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    scale_size(range = c(0.25, 5)) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Compound), 
               nrow = 3, scales = "fixed") +
    labs(y = "Allele frequency") +
    theme(legend.position = "top") +
    guides(colour = guide_legend(order = 1, nrow = 3), 
         shape = guide_legend(order = 2, nrow = 3),
         size = guide_legend(order = 3, nrow = 3))
     
```

#### Loperamide - zoom in peg.1562

Not very interesting for Loperamide, as the AF in parental and control strains is even higher than in Loperamide evolved populations. _This gene has been discussed in ALE 2.0 before, as the 5th TonB in 1,1912 Kb - 1, 945 Kb._

```{r peg1562, echo=FALSE, warning=FALSE, fig.width=10, fig.height=4}
  this_gene_ID <- c("peg.1562")
  this_gene <- all_others[all_others$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID %in% this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID %in% this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(1.4, 1), 
                    labels = c('a', 'b'))
```

#### Loperamide - zoom in peg.1703/1704

TrKAH units for K+ channel / transporter, might be related to the growth adaptation. Study showed that trkA mutation will depolarize the membrane comparing to wild type ([Zhang _et al._, 2020](https://www.nature.com/articles/s41467-019-14240-9)). Since these two genes are located on __- strand__, therefore, most of the frame shift mutations were at the beginning of the protein coding sequences.

```{r trkA_trkH, echo=FALSE, warning=FALSE, fig.width=10, fig.height=4}
  this_gene_ID <- c("peg.1703", "peg.1704")
  this_gene <- all_others[all_others$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID %in% this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID %in% this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio)) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(2, 1), 
                    labels = c('a', 'b'))
```

#### Loperamide - zoom in peg.1976

Variant (at position 2,420,681) locates after gene peg.1976 (Glyco_trans_1_4,Glyco_transf_4), but found in many samples -> probably not doing anything?

```{r peg1976, echo=FALSE, warning=FALSE, fig.width=10, fig.height=4}
  this_gene_ID <- c("peg.1976")
  this_gene <- all_others[all_others$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID %in% this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID %in% this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
   # geom_vline(xintercept = c(this_start, this_end), 
  #             linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio)) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(1, 2), 
                    labels = c('a', 'b'))
```

#### Loperamide - zoom in peg.2523

Synonymous variant in coding region of gene GDP-mannose 4,6-dehydratase.

```{r peg2523, echo=FALSE, warning=FALSE, fig.width=10, fig.height=4}
  this_gene_ID <- c("peg.2523")
  this_gene <- all_others[all_others$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID %in% this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID %in% this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
   # geom_vline(xintercept = c(this_start, this_end), 
  #             linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio)) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(1, 1.6), 
                    labels = c('a', 'b'))
```

#### Loperamide - zoom in peg.2890

Missense variant in coding region of gene putative type IIS restriction/modification enzyme.

```{r peg2890, echo=FALSE, warning=FALSE, fig.width=8, fig.height=4}
  this_gene_ID <- c("peg.2890")
  this_gene <- all_others[all_others$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID %in% this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID %in% this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio)) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(1.2, 1), 
                    labels = c('a', 'b'))
```

### Variants detected for compound Ezetimibe and Simvastatin (following Figure 1a)

Firstly, we checked the overall distribution of variants in compound and control populations.

```{r get_lma, echo = FALSE, warning=FALSE}
  lma <- all_others[all_others$Compound %in% c("DMSO_control", "NT5002",
                                              "Ezetimibe", "Simvastatin"), ]                                   
  lma_02 <- lma[lma$Ratio > 0.2, ]
```

Among those variants, we decided to focus on the areas where AF > 0.2 variants were detected, including coding area `r unique(lma_02$Gene)`. All those genes were discussed either in gene wise variant analysis or other compound wise variant analysis.  

```{r lma, echo=FALSE, warning=FALSE, fig.width=8, fig.height=9}
   ggplot(lma, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    scale_size(range = c(0.25, 5)) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Compound), 
               nrow = 4, scales = "fixed") +
    labs(y = "Allele frequency") +
    theme(legend.position = "top") +
    guides(colour = guide_legend(order = 1, nrow = 3), 
         shape = guide_legend(order = 2, nrow = 4),
         size = guide_legend(order = 3, nrow = 4)) 
     
```

### Variants detected for compound Xanthan gum (appeared several times in gene wise variant plots)

Firstly, we checked the overall distribution of variants in compound and control populations.

```{r get_xg, echo = FALSE, warning=FALSE}
  xg <- all_others[all_others$Compound %in% c("Water_control", "NT5002",
                                              "Xanthan_gum"), ]                                   
  xg_02 <- xg[xg$Ratio > 0.2, ]
```

Among those variants, we decided to focus on the areas where AF > 0.2 variants were detected, including coding area `r unique(xg_02$Gene)`. 

```{r xg, echo=FALSE, warning=FALSE, fig.width=8, fig.height=6}
   ggplot(xg, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio),
               alpha = 0.9) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    scale_size(range = c(0.25, 5)) +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Compound), 
               nrow = 4, scales = "fixed") +
    labs(y = "Allele frequency") +
    theme(legend.position = "top") +
    guides(colour = guide_legend(order = 1, nrow = 3), 
         shape = guide_legend(order = 2, nrow = 4),
         size = guide_legend(order = 3, nrow = 4)) 
     
```

#### Xanthan gum - zoom in peg.935

Gene peg.935 (RHS repeat-associated core domain protein) is on - strand. 

```{r peg935, echo=FALSE, warning=FALSE, fig.width=8, fig.height=4}
  this_gene_ID <- "peg.935"
  this_gene <- all_others[all_others$Gene == this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID == this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID == this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(1.4, 1), 
                    labels = c('a', 'b'))
```

#### Xanthan gum - zoom in peg.2423 and 2427

Genes peg.2423 (YjbH, -), peg.2424 (-), peg.2427 (YjbH, +, tr|R7EIH8|R7EIH8_9BACE, [lipoprotein](https://www.uniprot.org/uniprotkb/R7EIH8/entry)).

```{r peg2423/2424/2427, echo=FALSE, warning=FALSE, fig.width=7, fig.height=7}
  this_gene_ID <- c("peg.2423", "peg.2424", "peg.2427")
  this_gene <- all_others[all_others$Gene %in% this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID %in% this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID %in% this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "bottom") + 
    guides(colour = guide_legend(order = 1, nrow = 2), 
         shape = guide_legend(order = 2, nrow = 2)) +
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, nrow = 2, align = "h", axis = "b", 
                    rel_heights = c(1.4, 1), 
                    labels = c('a', 'b'))
```

#### Xanthan gum - zoom in peg.2855

Gene peg.2855 (Biotin carboxylase of acetyl-CoA carboxylase (EC 6.3.4.14);Ontology_term=KEGG_ENZYME:6.3.4.14) is on + strand. 

```{r peg2855, echo=FALSE, warning=FALSE, fig.width=8, fig.height=4}
  this_gene_ID <- "peg.2855"
  this_gene <- all_others[all_others$Gene == this_gene_ID, ]
  
  ## Get the coordinate of this gene
  this_start <- gff[gff$Gene_ID == this_gene_ID, ]$Start
  this_end <- gff[gff$Gene_ID == this_gene_ID, ]$End
  
  a <- ggplot(this_gene, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               alpha = 0.9, size = 3) + 
    scale_color_manual(values = concentration_color) +
    labs(x = "Position") +
    geom_vline(xintercept = c(this_start, this_end), 
               linetype = "dashed", color = "salmon") +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    theme(legend.position = "right") + 
    labs(y = "Allele frequency") +
    main_theme 

  b <- ggplot(this_gene, aes(Compound, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect),
               size = 3) + 
    main_theme +
    scale_color_manual(values = concentration_color) +
    scale_shape_manual(labels = t_shape, values = c_shape) +
    labs(y = "Allele frequency") +
    theme(legend.position = "none",
          axis.text.x = element_text(size = 10, angle = 90, 
                                     vjust = 0.5, hjust=1))  
  plot_grid(a, b, ncol = 2, align = "h", axis = "b", 
                    rel_widths = c(1.4, 1), 
                    labels = c('a', 'b'))
```



### Variants with AF > 0.1 [[how to decide the cutoff??]]

We also get variants with AF > 0.1 and compared them between compounds -> PCoA of these variants? Rows are populations, columns are variants?

```{r cmp_wise_01, echo = FALSE, warning = FALSE, fig.height=24, fig.width=20}

  all_01 <- all_others[all_others$Ratio > 0.1, ]

  ggplot(all_01, aes(POS, Ratio)) +
    geom_point(aes(color = Concentration, shape = Effect, size = Ratio)) + 
    scale_shape_manual(labels = t_shape, values = c_shape) +
    main_theme +
    geom_hline(yintercept = 0.1, 
               linetype = "dashed", color = "gold") +
    scale_color_manual(values = concentration_color) +
    facet_wrap(~factor(Compound), 
               ncol = 4, scales = "free_y") +
    labs(y = "Allele frequency") +
    theme(legend.position = "top")
```

